Drivers/Periphs/Src/SDCard.c

```diff
@@ -76,59 +99,91 @@ FRESULT SDCard_GetStatistics() {
  * @return BaseType_t - pdTrue if placed, errQUEUE_FULL if full
  */
 BaseType_t SDCard_PutInQueue(SDCard_t* data) {
-    return xQueueSendToBack(SDCardQ, data, (TickType_t)0);
+    BaseType_t success = xQueueSendToBack(SDCardQ, data, (TickType_t)0);
+    if (success == errQUEUE_FULL) {
+        SDCDroppedMessages++;
+    }
+    return success;
 }
 
 /**
- * @brief Formats data to be written to SD card based on type of data input. 
- * NOTE: Non-Blocking - Returns error if there is no data 
+ * @brief Formats data and writes to SD card based on type of data input. 
+ *        Data is buffered and written in large chunks. 
+ *        !!! DOES NOT SYNC DATA !!! You must call SDCard_SyncLogFiles() to save.
+ * @note: Non-Blocking - Returns error if there is no data 
  * @param none
- * @return FRESULT FR_OK if ok and other errors specified in ff.h
+ * @return FRESULT FR_OK if ok, FR_DISK_ERR if SD queue is empty, and other errors specified in ff.h
  */
 FRESULT SDCard_Sort_Write_Data(){
     // check if data from queue is from imu, gps, or can.
     // send data to corresponding file in sd card
     SDCard_t cardData;
-    enum filenames_idx {CAN_FNAME = 0, IMU_FNAME, GPS_FNAME};
-    static char message[SDCARD_WRITE_BUFSIZE];
+    FRESULT success = FR_OK;
+
+    if (xQueueReceive(SDCardQ, &cardData, (TickType_t)1) != pdTRUE) return SDC_QUEUE_EMPTY;
+    if (((uint32_t)cardData.id) >= LARGEST_SDC_ID) return FR_DISK_ERR;
```

Comment:
TBH, not sure how this means that the queue is empty.

Reply:
If queue receive returns not true then it's empty

Reply:
Comment says FR_DISK_ERR means queue is empty